pages/mdbook.yml

```
@@ -29,14 +29,11 @@ jobs:
   build:
     runs-on: ubuntu-latest
     env:
-      MDBOOK_VERSION: 0.4.36
+      MDBOOK_VERSION: 0.4.40
     steps:
       - uses: actions/checkout@v4
       - name: Install mdBook
-        run: |
-          curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf -y | sh
-          rustup update
-          cargo install --version ${MDBOOK_VERSION} mdbook
+        run: cargo install mdbook@${MDBOOK_VERSION}
```

Basically this can even completely skip building mdbook itself, and downloading the binary release instead:

```suggestion
        run: curl -sSL "https://github.com/rust-lang/mdBook/releases/download/v$MDBOOK_VERSION/mdbook-v$MDBOOK_VERSION-x86_64-unknown-linux-gnu.tar.gz" | tar -xz --directory /home/runner/.cargo/bin
```

that's instantaneous, taking down the run time from over two minutes to mere seconds…

(I just wasn't sure whether that's not "too fragile" whereas the cargo install is more foolproof, as long as the tools are in the runner…)